typescript/infra/src/utils/rpcUrls.ts

```diff
@@ -50,28 +61,167 @@ export async function setAndVerifyRpcUrls(
-      )}\n`,
-    );
+    return [];
+  }
+
+  const currentSecrets = await getSecretRpcEndpoints(environment, chain);
+  console.log(
+    `Current secrets found for ${chain} in ${environment} environment:\n${formatRpcUrls(
+      currentSecrets,
+    )}\n`,
+  );
+  return currentSecrets;
+}
+
+// Copied from @inquirer/prompts as it's not exported :(
+type Choice<Value> = {
+  value: Value;
+  name?: string;
+  description?: string;
+  short?: string;
+  disabled?: boolean | string;
+  type?: never;
+};
+
+/**
+ * Prompt the user to input RPC URLs for the given chain.
+ * The user can choose to input new URLs, use all registry URLs, or use existing URLs
+ * from secrets or the registry.
+ * @param chain The chain to input RPC URLs for
+ * @param existingUrls The existing RPC URLs for the chain
+ * @returns The selected RPC URLs
+ */
+async function inputRpcUrls(
+  chain: string,
+  existingUrls: string[],
+): Promise<string[]> {
+  const selectedUrls: string[] = [];
+
+  const registryUrls = getChain(chain).rpcUrls.map((rpc) => rpc.http);
+
+  const existingUrlChoices: Array<Choice<string>> = existingUrls.map(
+    (url, i) => {
+      return {
+        value: url,
+        name: `${url} (existing index ${i})`,
+      };
+    },
+  );
+  const registryUrlChoices: Array<Choice<string>> = registryUrls.map(
+    (url, i) => {
+      return {
+        value: url,
+        name: `[PUBLIC] ${url} (registry index ${i})`,
+      };
+    },
+  );
+
+  enum SystemChoice {
+    ADD_NEW = 'Add new RPC URL',
+    DONE = 'Done',
+    USE_REGISTRY_URLS = 'Use all registry URLs',
+    REMOVE_LAST = 'Remove last RPC URL',
   }
+
+  const pushSelectedUrl = async (newUrl: string) => {
+    const providerHealthy = await testProvider(chain, newUrl);
+    if (!providerHealthy) {
+      const yes = await confirm({
+        message: `Provider at ${newUrl} is not healthy. Do you want to continue adding it?\n`,
+      });
+      if (!yes) {
+        console.log('Skipping provider');
+        return;
+      }
+    }
+    selectedUrls.push(newUrl);
+  };
+  const separator = new Separator('-----');
+
+  while (true) {
+    console.log(`Selected RPC URLs: ${formatRpcUrls(selectedUrls)}\n`);
+
+    // Sadly @inquirer/prompts doesn't expose the types needed here
+    const choices: (Separator | Choice<any>)[] = [
+      ...[SystemChoice.DONE, SystemChoice.ADD_NEW].map((choice) => ({
+        value: choice,
+      })),
+      {
+        value: SystemChoice.USE_REGISTRY_URLS,
+        name: `Use registry URLs (${JSON.stringify(registryUrls)})`,
+      },
+      separator,
+      ...existingUrlChoices,
+      separator,
+      ...registryUrlChoices,
+    ];
+    if (selectedUrls.length > 0) {
+      choices.push(separator);
+      choices.push({
+        value: SystemChoice.REMOVE_LAST,
+      });
+    }
+
+    const selection = await select({
+      message: 'Select RPC URL',
+      choices,
+      pageSize: 30,
+    });
+
+    if (selection === SystemChoice.DONE) {
```

Comment:
a "select all" would be nice so i don't have to manually select each registry one on new chain deploys